Sources/Call.swift

```diff
@@ -44,5 +46,81 @@ open class Call {
 
         return rootedJson[rootNode]
     }
-
+    
+    private func insertParameters(request: URLRequest) -> URLRequest {
+        if (self.parameters == nil) {
+            return request
+        }
+        do {
+            switch (self.parameters!.type!) {
+            case .httpHeader:
+                guard let headers = self.parameters?.parameters as? [String: String] else {
+                    throw FaroError.malformed(info: "HTTP headers must be in a [String: String] format")
+                }
+                return insert(headers: headers, request: request)
+            case .urlComponents:
+                guard let componentsDict = self.parameters?.parameters as? [String: String] else {
+                    throw FaroError.malformed(info: "URL components must first be cast to strings")
+                }
+                return insert(componentsDict: componentsDict, request: request)
+            case .jsonBody:
+                return insert(json: (self.parameters?.parameters)!, request: request)
+            }
+        } catch {
+            printError(error)
+            return request
+        }
+    }
+    
+    private func insert(headers: [String: String], request: URLRequest) -> URLRequest {
+        var newRequest = request
+        for (key, value) in headers {
+            newRequest.addValue(value, forHTTPHeaderField: key)
+        }
+        return newRequest
+    }
+    
+    private func insert(componentsDict: [String: String], request: URLRequest!) -> URLRequest {
+        var newRequest: URLRequest! = request
+        var components = URLComponents(url: newRequest.url!, resolvingAgainstBaseURL: false)
+        if (components?.queryItems == nil) {
+            components?.queryItems = [URLQueryItem]()
+        }
+        for (key, value) in componentsDict {
+            components?.queryItems?.append(URLQueryItem(name: key, value: value))
+        }
+        newRequest.url = components?.url
+        return newRequest
+    }
+    
+    private func insert(json: [String: Any], request: URLRequest!) -> URLRequest {
```

Comment:
💣 Kan je deze functie ook niet wat opsplitsen in kleinere functies. Nu is het nogal veel om te processen.